impeller/entity/contents/linear_gradient_contents.cc

```diff
@@ -68,25 +72,67 @@ bool LinearGradientContents::Render(const ContentContext& renderer,
                    entity.GetTransformation();
   frame_info.matrix = GetInverseMatrix();
 
-  FS::GradientInfo gradient_info;
-  gradient_info.start_point = start_point_;
-  gradient_info.end_point = end_point_;
-  gradient_info.start_color = colors_[0].Premultiply();
-  gradient_info.end_color = colors_[1].Premultiply();
-  gradient_info.tile_mode = static_cast<Scalar>(tile_mode_);
-
-  Command cmd;
-  cmd.label = "LinearGradientFill";
-  cmd.pipeline = renderer.GetLinearGradientFillPipeline(
-      OptionsFromPassAndEntity(pass, entity));
-  cmd.stencil_reference = entity.GetStencilDepth();
-  cmd.BindVertices(
-      vertices_builder.CreateVertexBuffer(pass.GetTransientsBuffer()));
-  cmd.primitive_type = PrimitiveType::kTriangle;
-  FS::BindGradientInfo(
-      cmd, pass.GetTransientsBuffer().EmplaceUniform(gradient_info));
-  VS::BindFrameInfo(cmd, pass.GetTransientsBuffer().EmplaceUniform(frame_info));
-  return pass.AddCommand(std::move(cmd));
+  if (colors_.size() > 2) {
```

Comment:
nit: consider moving the code in the if/else here to separate functions, they're getting a bit long.

Reply:
Also, here or perhaps where you define the `FillTwoColor` pipelines, consider adding a comment about why -e.g. "most gradients use only two colors/stops and we can trivially avoid the render target switch needed to create the texture for multi-color gradients".

I'm still curious what Chinmay thinks of this though.